### PR TITLE
Update test uart types

### PR DESCRIPTION
**Pull request by tianxiemaochiyu**

### CODE REVIEW

File: packages/apps-config/src/api/spec/uart.ts [hunk 1]
```diff
@@ -21,17 +21,20 @@ const definitions: OverrideBundleDefinition = {
...
+        ProposalId: 'u64',
+        TokenBalance: 'Balance',
         LotteryKind: {
-          _enum: {
-            Routine: 'Null',
-            TreasuryFunded: 'ReasonIndex'
-          }
+          _enum: [
+            'Routine',
+            'TreasuryFunded(ReasonIndex)'
```
Comment (jacogr) [inline packages/apps-config/src/api/spec/uart.ts:30]: The original enum here was correct, `TreasuryFunded(ReasonIndex)` is not a valid type.

File: packages/apps-config/src/api/spec/uart.ts [hunk 2]
```diff
@@ -21,17 +21,20 @@ const definitions: OverrideBundleDefinition = {
...
+          ]
         },
         LottoResult: {
-          _enum: {
-            Routine: '(AccountId, Balance)',
-            TreasuryFunded: 'Balance'
-          }
+          _enum: [
+            'Routine(AccountId, Balance)',
+            'TreasuryFunded(Balance)'
```
Comment (jacogr) [inline packages/apps-config/src/api/spec/uart.ts:36]: As above, the original types was correct, this makes it invalid (or rather a c-like enum which won't be able to decode the values)

File: packages/apps-config/src/api/spec/uart.ts [hunk 3]
```diff
@@ -159,8 +201,9 @@ const definitions: OverrideBundleDefinition = {
         },
         ReFungibleItemType: {
           Collection: 'u64',
-          Owner: 'Vec<Ownership>',
-          Data: 'Vec<u8>'
+          Owner: 'Vec<Ownership<AccountId>>',
```
Comment (jacogr) [inline packages/apps-config/src/api/spec/uart.ts:204]:
```suggestion
          Owner: 'Vec<Ownership>',
```

Unlike on the Rust side, the types parser is not clever enough to be passed a generic parameter.

File: packages/apps-config/src/api/spec/uart.ts (hunk 2 above)
Comment (tianxiemaochiyu) [reply packages/apps-config/src/api/spec/uart.ts:36]: ok

**Comment by tianxiemaochiyu** [comment]
@jacogr Thanks for the help.Updated.

File: packages/apps-config/src/api/spec/uart.ts (hunk 1 above)
Comment (jacogr) [inline packages/apps-config/src/api/spec/uart.ts:30]: The enums are still incorrect, the correct form is indeed as it was in the original.

```js
Routine: 'Null', // this one is has not extra data
TreasuryFunded: 'ReasonIndex' // This one contains data
```

As mentioned originally, this affects the `LotteryKind` (original correct, new version is not), `LotteryResult` (same where original is correct, new version is incorrect as will match as a c-like enum)

**Comment by tianxiemaochiyu** [comment]
@jacogr Sorry,updated.

**Review by jacogr — APPROVED** [review]

**Pull request merged**